module/zfs/zfs_replay.c

```diff
@@ -643,19 +643,33 @@ zfs_replay_link(void *arg1, void *arg2, boolean_t byteswap)
 }
 
 static int
-zfs_replay_rename(void *arg1, void *arg2, boolean_t byteswap)
+do_zfs_replay_rename(void *arg1, void *arg2, boolean_t byteswap,
+    uint64_t rflags)
 {
 	zfsvfs_t *zfsvfs = arg1;
 	lr_rename_t *lr = arg2;
 	char *sname = (char *)(lr + 1);	/* sname and tname follow lr_rename_t */
 	char *tname = sname + strlen(sname) + 1;
 	znode_t *sdzp, *tdzp;
-	int error;
-	int vflg = 0;
+	int error, vflg = 0;
 
 	if (byteswap)
 		byteswap_uint64_array(lr, sizeof (*lr));
 
+	/* Only Linux currently supports RENAME_* flags. */
+#ifdef __linux__
+	VERIFY0(rflags & ~(RENAME_EXCHANGE | RENAME_WHITEOUT));
+
+	IMPLY(rflags & RENAME_EXCHANGE,
```

Comment:
To my reading of `IMPLY`'s definition, this will be a no-op in release builds. Are you sure you want this? https://sourcegraph.com/github.com/openzfs/zfs/-/blob/include/os/linux/spl/sys/debug.h?L126-141

(Maybe we should have a VERIFY_IMPLY macro?)

Reply:
You're right, this needs to be `VERIFY_IMPLY` (I'll add that).